### DISCUSSION TITLE
VisualShader: Add ability to close "Create Shader Node" window by clicking on anywhere on screen

### DISCUSSION THREAD

**Discussion by Foyezes** [godotengine/godot-proposals#11203] [GUI]
Sometimes if I accidentally press right click and open the "Create Shader Node" window, I have to drag my mouse to the minimize button. If there was a system to close the window by clicking on the spaces outside the window that would be super helpful.

Also being able to open a node's setting by directly right clicking on the node directly, instead of having to select the node first will be good. Currently if I rick click on a node directly it opens the "Create shader node" window.
![image](https://github.com/user-attachments/assets/25f89557-6ef5-411d-af3b-9cdb6f4a1b18)
![image](https://github.com/user-attachments/assets/540952b8-e324-45c0-8912-6ee2607124a7)

**Comment by tetrapod00** [comment]
I very much support being able to close the Add Nodes dialog by clicking anywhere outside it.

I did a quick search and couldn't find a full proposal for it, so I'd suggest making one, so more people can see it and it can start collecting feedback and support. It's not a proposal that requires much complexity or knowledge of how it would be implemented, IMO.

**Comment by Foyezes** [comment]
Proposal: https://github.com/godotengine/godot-proposals/issues/11213

**Reply by tetrapod00** [reply]
Thanks! Since it's a full proposal now, I'm going to close this discussion in favor of the proposal.
